polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -181,6 +181,12 @@ impl<H: Copy> From<CandidateDescriptorV2<H>> for CandidateDescriptor<H> {
 	}
 }
 
+impl<H: Encode + Decode + Copy> From<CandidateDescriptor<H>> for CandidateDescriptorV2<H> {
+	fn from(value: CandidateDescriptor<H>) -> Self {
+		Decode::decode(&mut value.encode().as_slice()).unwrap()
```

Comment:
I think we could probably do it faster than encode/decode. But I see it's only being used in tests so it's fine